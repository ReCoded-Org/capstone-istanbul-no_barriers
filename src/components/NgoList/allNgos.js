import image from "./image/image.jpg";

// This will be a fetch from the database
const allNgos = [
  {
    id: "firebaseDocIda",
    image,
    name: "Teng",
    service: "Health Care",
    city: "Ankara",
    rating: "4",
    details: {
      facilityName: "Teng clinic",
      address: "Çankaya/Ankara",
      serviceDescription:
        "Suspendisse pulvinar pretium scelerisque. Suspendisse potenti. Sed ullamcorper enim sed lacinia sagittis. Nulla tellus risus, maximus et vestibulum id, tempor eget ipsum. Praesent convallis dictum turpis eu ullamcorper. Curabitur in ultrices sapien. Etiam lacinia ante quis luctus luctus. Nullam efficitur turpis orci, at bibendum tortor efficitur sed. Nam accumsan dolor luctus blandit lobortis.",
      responsiblePerson: "Ali Demir",
      phoneNumber: "+90 000 000 00",
      email: "tengAdmin@teng.com",
      website: "www.teng.com",
    },
  },
  {
    id: "firebaseDocIdb",
    image,
    name: "Teng",
    service: "Health Care",
    city: "Ankara",
    rating: "4",
    details: {
      facilityName: "Teng clinic",
      address: "Çankaya/Ankara",
      serviceDescription:
        "Suspendisse pulvinar pretium scelerisque. Suspendisse potenti. Sed ullamcorper enim sed lacinia sagittis. Nulla tellus risus, maximus et vestibulum id, tempor eget ipsum. Praesent convallis dictum turpis eu ullamcorper. Curabitur in ultrices sapien. Etiam lacinia ante quis luctus luctus. Nullam efficitur turpis orci, at bibendum tortor efficitur sed. Nam accumsan dolor luctus blandit lobortis.",
      responsiblePerson: "Ali Demir",
      phoneNumber: "+90 000 000 00",
      email: "tengAdmin@teng.com",
      website: "www.teng.com",
    },
  },
  {
    id: "firebaseDocIdc",
    image,
    name: "ABC",
    service: "Education",
    city: "Şanlıurfa",
    rating: "4",
    details: {
      facilityName: "ABC School",
      address: "Şanlıurfa",
      serviceDescription:
        "Suspendisse pulvinar pretium scelerisque. Suspendisse potenti. Sed ullamcorper enim sed lacinia sagittis. Nulla tellus risus, maximus et vestibulum id, tempor eget ipsum. Praesent convallis dictum turpis eu ullamcorper. Curabitur in ultrices sapien. Etiam lacinia ante quis luctus luctus. Nullam efficitur turpis orci, at bibendum tortor efficitur sed. Nam accumsan dolor luctus blandit lobortis.",
      responsiblePerson: "Aişe Şen",
      phoneNumber: "+90 000 000 00",
      email: "abcAdmin@abc.com",
      website: "www.abc.com",
    },
  },
];

export default allNgos;
