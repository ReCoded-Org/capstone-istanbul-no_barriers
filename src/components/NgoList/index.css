.ngoLogo {
  height: 4rem;
  width: 4.1rem;
  border-radius: 50%;
  border: 1px solid blue;
}
