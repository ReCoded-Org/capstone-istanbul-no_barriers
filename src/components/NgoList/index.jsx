import React, { useState } from "react";
import { Row, Col, Button, Collapse } from "react-bootstrap";
import allNgos from "./allNgos";
import "./index.css";
import NgoDetails from "../NgoDetails";

// This renders the NGOs which are filtered or searched
// It renders all NGOs by default if a parameter is not given
const NgoList = ({ ngos = allNgos }) => {
  const [openDetails, setOpenDetails] = useState(false);

  return (
    <div>
      <Row className="ngoListTitles d-flex justify-content-around align-items-center">
        <Col>&nbsp;</Col>
        <Col>NGO Name</Col>
        <Col>Service</Col>
        <Col>City</Col>
        <Col>Rating</Col>
        <Col>&nbsp;</Col>
      </Row>
      <div className="tablesBody">
        {ngos.map((ngo, index) => {
          if (index % 2 === 0) {
            return (
              <div className="tableNgoInfo" key={ngo.id}>
                <Row>
                  <Col>
                    <img src={ngo.image} alt="NGO Logo" className="ngoLogo" />
                  </Col>
                  <Col>{ngo.name}</Col>
                  <Col>{ngo.service}</Col>
                  <Col>{ngo.city}</Col>
                  <Col>{ngo.rating}</Col>
                  <Col>
                    <Button
                      onClick={() => setOpenDetails(!openDetails)}
                      aria-expanded={openDetails}
                      aria-controls={`${ngo.id}`}
                    >
                      Details
                    </Button>
                  </Col>
                </Row>
                <Collapse in={openDetails}>
                  <Row
                    className="ngoDetails d-flex justify-content-around"
                    id={`${ngo.id}`}
                  >
                    <Col md={6}>
                      <NgoDetails details={ngo.details} />
                    </Col>
                  </Row>
                </Collapse>
              </div>
            );
          }
          return (
            <div className="tableNgoInfo" key={ngo.id}>
              <Row>
                <Col>
                  <img src={ngo.image} alt="NGO Logo" className="ngoLogo" />
                </Col>
                <Col>{ngo.name}</Col>
                <Col>{ngo.service}</Col>
                <Col>{ngo.city}</Col>
                <Col>{ngo.rating}</Col>
                <Col>
                  <Button
                    onClick={() => setOpenDetails(!openDetails)}
                    aria-expanded={openDetails}
                    aria-controls={`${ngo.id}`}
                  >
                    Details
                  </Button>
                </Col>
              </Row>
              <Collapse in={openDetails}>
                <Row
                  className="ngoDetails d-flex justify-content-around"
                  id={`${ngo.id}`}
                >
                  <Col md={6}>
                    <NgoDetails details={ngo.details} />
                  </Col>
                </Row>
              </Collapse>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default NgoList;
