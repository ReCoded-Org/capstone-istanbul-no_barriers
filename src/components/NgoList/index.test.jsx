import React from "react";
import renderer from "react-test-renderer";
import NgoList from "./index";
import allNgos from "./allNgos";

it("NgoList renders correctly", () => {
  const tree = renderer.create(<NgoList ngos={allNgos} />).toJSON();
  expect(tree).toMatchSnapshot();
});
