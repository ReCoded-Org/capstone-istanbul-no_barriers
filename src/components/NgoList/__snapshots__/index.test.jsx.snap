// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgoList renders correctly 1`] = `
<div
  className="ngoList"
>
  <table
    className="table-borderless ngoListTable table"
  >
    <thead>
      <tr>
        <th
          className="logoCol"
          scope="col"
        >
           
        </th>
        <th
          scope="col"
        >
          resources.AdvancedFilters.ngoName
        </th>
        <th
          scope="col"
        >
          resources.AdvancedFilters.service
        </th>
        <th
          scope="col"
        >
          resources.AdvancedFilters.city
        </th>
        <th
          scope="col"
        >
          resources.AdvancedFilters.rating
        </th>
        <th
          scope="col"
        >
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="table-info"
      >
        <td
          className="d-table-cell align-middle"
        >
          <img
            align="right"
            alt="NGO Logo"
            className="ngoLogo"
            src="image.jpg"
          />
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Teng
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Health Care
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Ankara
        </td>
        <td
          className="d-table-cell align-middle"
        >
          4
        </td>
        <td
          className="d-table-cell align-middle btnCol"
        >
          <button
            aria-controls="undefined"
            className="btnMoreDetails shadow-none btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            resources.ngoList.buttonOpenDetails
          </button>
        </td>
      </tr>
      <tr
        className="table-light"
      >
        <td
          className="d-table-cell align-middle"
        >
          <img
            align="right"
            alt="NGO Logo"
            className="ngoLogo"
            src="image.jpg"
          />
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Teng
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Health Care
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Ankara
        </td>
        <td
          className="d-table-cell align-middle"
        >
          4
        </td>
        <td
          className="d-table-cell align-middle btnCol"
        >
          <button
            aria-controls="undefined"
            className="btnMoreDetails shadow-none btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            resources.ngoList.buttonOpenDetails
          </button>
        </td>
      </tr>
      <tr
        className="table-info"
      >
        <td
          className="d-table-cell align-middle"
        >
          <img
            align="right"
            alt="NGO Logo"
            className="ngoLogo"
            src="image.jpg"
          />
        </td>
        <td
          className="d-table-cell align-middle"
        >
          ABC
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Education
        </td>
        <td
          className="d-table-cell align-middle"
        >
          Şanlıurfa
        </td>
        <td
          className="d-table-cell align-middle"
        >
          4
        </td>
        <td
          className="d-table-cell align-middle btnCol"
        >
          <button
            aria-controls="undefined"
            className="btnMoreDetails shadow-none btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            resources.ngoList.buttonOpenDetails
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
